### PR TITLE
Add missing autofeatures to resolveAsSet output

### PR DESCRIPTION
**Pull request by TayyabDev**

### CODE REVIEW

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ResolveDirector.java [hunk 1]
```diff
@@ -641,8 +641,24 @@ void resolveAutoFeatures(Collection<ESAAsset> autoFeatures, ArrayList<InstallAss
...
+                Set<String> returnedFeatureList = new HashSet<>();
+                for(List<RepositoryResource>  resList  : installResources){
+                    for(RepositoryResource res : resList) {
+                        if (res.getType().equals(ResourceType.FEATURE)) {
+                            EsaResource esa = (EsaResource) res;
+                            returnedFeatureList.add(esa.getProvideFeature());
+                        }
+                    }
+                }
+                InstallUtils.setResolveAsSetFeatures(returnedFeatureList);
```
Comment (alvinso) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ResolveDirector.java]: You should skip the 2nd call to resolve the auto features if returnedFeatureList is empty.

**Comment by TayyabDev** [comment]
#build

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ResolveDirector.java (hunk 1 above)
Comment (TayyabDev) [reply dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/ResolveDirector.java]: fixed in commit https://github.com/OpenLiberty/open-liberty/pull/11203/commits/54588424ec43750b0f0e27d631b3155647c611a4

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#build

File: dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java [hunk 2]
```diff
@@ -738,7 +738,6 @@ private static boolean containsIgnoreCase(Collection<String> featureToInstall, S
 
             resolver = new RepositoryResolver(productDefinitions, installedFeatures, Collections.<IFixInfo> emptySet(), repoList);
             resolveResult = resolver.resolveAsSet((Collection<String>) data.get(FEATURES_TO_RESOLVE));
-
```
Comment (alvinso) [inline dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java]: can we call ResolveDirector.resolveAutoFeatures right after the resolveAsSet call?

**Comment by TayyabDev** [comment]
#build

**Comment by TayyabDev** [comment]
#run-libby-bot

**Pull request merged**